File: contests/abs/abc088_b.ml
open Core

let read_line () = In_channel.(input_line_exn stdin)

let () =
  let _ = read_line () in
  let a = read_line () |> String.split ~on:' ' |> List.map ~f:int_of_string in
  List.(
    sort ~compare:Int.descending a
    |> mapi ~f:(fun i x -> (i, x))
    |> partition_tf ~f:(fun (i, _) -> i % 2 = 0)
    |> fun (a, b) ->
    let snd_sum = sum (module Int) ~f:snd in
    snd_sum a - snd_sum b)
  |> printf "%d\n"

File: contests/abs/abc088_b.fs
stdin.ReadLine() |> ignore

stdin.ReadLine().Split()
|> Array.map int64
|> Array.sortDescending
|> Array.fold (fun (l, r) x -> x + r, l) (0L, 0L)
||> (-)
|> abs
|> printfn "%d"
